src/lib.rs

```diff
@@ -489,8 +489,55 @@ where
         transactional_bonsai_storage: BonsaiStorage<ChangeID, DB::Transaction, H>,
     ) -> Result<(), BonsaiStorageError<<DB as BonsaiPersistentDatabase<ChangeID>>::DatabaseError>>
     {
-        self.tries
-            .db_mut()
-            .merge(transactional_bonsai_storage.tries.db())
+        // Applying changes (all cache_leaf_modified) before calling
+        //self.tries.db_mut().merge() work for all test but one
+        //(merge_with_uncommitted_remove, it fails with RocksDB.Busy error,
+        //which seems to be related to use of OptimisticTransactions, but
+        //removing them does not solve this particular problem but causes
+        //others).
+
+        //Applying changes after call to merge() would imply a big refactor as
+        //merge take ownership of  its arguments, hence changes are not
+        //available anymore.
+
+        //Hence the solution which is a tradeoff between the two solutions:
+        //1. memorize changes
+        //2. merge tries
+        //3. apply changes
```

Comment:
fmt comment

Reply:
will remove, the code is self-explanatory